text/0431-sagemaker-l2-endpoint.md

```diff
@@ -0,0 +1,1150 @@
+launched the following features which would require further additions to the L2 API contracts:
+
+* [Multi-model endpoints][multi-model]
+* [Model Monitor][model-monitor]
+* [Asynchronous inference][async-inference]
+* [Deployment guardrails][deployment-guardrails]
+* [Serverless inference][serverless-inference]
+
+Although some of these changes would be small and additive (e.g., `DataCaptureConfig` for Model
+Monitor), features like asynchronous and serverless inference represent more significant shifts in
+functionality. For example, SageMaker hosts real-time inference endpoints on EC2 instances, meaning
+that CloudWatch alarms and Application Auto Scaling rules operate on instance-based metrics. In
+contrast, serverless inference does not expose any instance-based metrics nor does it yet support
+auto-scaling. Since both features are specified via the CloudFormation resource
+`AWS::SageMaker::EndpointConfig`, the current recommendation of this RFC would be to support the
+specification of both use-cases through a single L2 `EndointConfig` construct. However, this
+presents a challenge when modeling helper APIs like `metricCPUUtilization` or
+`autoScaleInstanceCount` on a related construct as those methods would not universally apply to
+*all* endpoint types.
+
+Given that (1) RFC reviewers may have idiomatic recommendations to solve such modeling challenges
+and (2) the current proposed constructs are still viable for creating a SageMaker endpoint, the
+first draft of this RFC is being published without further revisions accommodating the above newer
+SageMaker features.
+
+[original-pr]: https://github.com/aws/aws-cdk/pull/6107
+[multi-model]: https://aws.amazon.com/blogs/machine-learning/save-on-inference-costs-by-using-amazon-sagemaker-multi-model-endpoints/
+[model-monitor]: https://aws.amazon.com/about-aws/whats-new/2019/12/introducing-amazon-sagemaker-model-monitor/
+[async-inference]: https://aws.amazon.com/about-aws/whats-new/2021/08/amazon-sagemaker-asynchronous-new-inference-option/
+[deployment-guardrails]: https://aws.amazon.com/about-aws/whats-new/2021/11/new-deployment-guardrails-amazon-sagemaker-inference-endpoints/
+[serverless-inference]: https://aws.amazon.com/about-aws/whats-new/2021/12/amazon-sagemaker-serverless-inference/
+
+### What is the high-level project plan?
+
+As the proposed design has been fully implemented in
+[CDK PR #20113](https://github.com/aws/aws-cdk/pull/20113), the delivery timeline of the
+implementation of this RFC will be contingent upon the scope of changes requested by reviewers. For
+baking, the L2 constructs for this module would be marked as experimental, leaving room for further
+adjustments prior to marking the APIs as stable.
+
+### Are there any open issues that need to be addressed later?
+
+1. Please see the [drawbacks section above](#what-are-the-drawbacks-of-this-solution) for potential
+   follow-on work (assuming it need not be incorporated into this RFC).
```

Comment:
I think we at least need an assessment of how leaving out that work could potentially create one way doors or impact our ability to add the feature set not included in this RFP.